morpheus/utils/databricks_utils.py

```diff
@@ -0,0 +1,44 @@
+import json
+import os
+from pathlib import Path
+import logging
+
+logger = logging.getLogger(__name__)
+
+def configure_databricks_connect(databricks_host: str,
+                                  databricks_token: str,
+                                  databricks_cluster_id: str,
+                                  databricks_port: str,
+                                  databricks_org_id: str):
+    if (os.environ.get('DATABRICKS_HOST', None) is None and databricks_host is None):
+        raise Exception("Parameter for databricks host not provided")
+    if (os.environ.get('DATABRICKS_TOKEN', None) is None and databricks_token is None):
+        raise Exception("Parameter for databricks token not provided")
+    if (os.environ.get('DATABRICKS_CLUSTER_ID', None) is None and databricks_cluster_id is None):
+        raise Exception("Parameter for databricks cluster not provided")
+
+    config_file = f"{str(Path.home())}/.databricks-connect"
+    add_config = False
+    
+    config = {
+        "host": os.environ.get('DATABRICKS_HOST', databricks_host),
+        "token": os.environ.get('DATABRICKS_TOKEN', databricks_token),
+        "cluster_id": os.environ.get('DATABRICKS_CLUSTER_ID', databricks_cluster_id),
+        "org_id": databricks_org_id,
+        "port": databricks_port
+    }
+    # check if the config file for databricks connect already exists
+    if os.path.exists(config_file):
+        # check if the config being added already exists, if so do nothing
+        with open(config_file) as f:
+            if json.dumps(config) in f.read():
+                logger.info("Configuration for databricks-connect already exists, nothing added!")
+            else:
+                logger.info("Configuration not found for databricks-connect, adding provided configs!")
+                add_config = True
+    else:
+        add_config = True
+    if add_config:
+        with open(config_file, "w+") as f:
+            f.write(json.dumps(config))
+        logger.info("Databricks-connect successfully configured!")
```

Comment:
We won't be calling this function very often, so eliminating unnecessary writes isn't something we need to worry about. With that considered, we can simplify this code by removing the file read and comparison.
```suggestion
    with open(config_file, "w") as fh:
        fh.write(json.dumps(config))
    logger.info("Databricks-connect successfully configured!")
```

Reply:
Can you be more specific? From what I'm reading, it appears that the logic is "if the file exists and contains the contents we would write, do not write the file. Otherwise, write the file." I don't see how there are two different sets of credentials in the config. Did you mean to append to the file with `a` instead of overwrite it with `w+`? If so, I think this logic needs some more work, as simply appending could lead to more problems.

Reply:
@cwharris the condition and other changes that were missed to be checked in should come in during next push, but the process is to check if the configuration exists, then we skip writing or making changes to the file else, we append the configs to the file

Reply:
Changes with `a` has been made, i'm testing the overall function to check for issues, will push in changes once its been tested

Reply:
What is the format of the configuration? I imagine it would be json, but the way you are appending configs makes it seem as though the format is json-lines. If it is not json-lines, this will present a deserialization problem if more than one config is added to the file.

Reply:
@cwharris this config is removed with latest changes, resolving the convo